apps/src/redux/parentalPermissionRequestReducer.ts

```diff
@@ -0,0 +1,182 @@
+
+const REQUEST_PARENTAL_PERMISSION_URL =
+  '/policy_compliance/child_account_consent';
+const REQUEST_PARENTAL_PERMISSION_PERFORM = 'requestParentalPermission/perform';
+export const REQUEST_PARENTAL_PERMISSION_SUCCESS =
+  'requestParentalPermission/success';
+const REQUEST_PARENTAL_PERMISSION_FAILURE = 'requestParentalPermission/failure';
+
+interface ParentalPermissionRequest {
+  parent_email: string;
+  requested_at: string;
+  resends_sent: number;
+  consent_status: string;
+}
+
+type ParentalPermissionRequestAction =
+  | {
+      type: typeof FETCH_PENDING_PERMISSION_REQUEST_PERFORM;
+    }
+  | {
+      type: typeof FETCH_PENDING_PERMISSION_REQUEST_SUCCESS;
+      parentalPermissionRequest: ParentalPermissionRequest | null;
+    }
+  | {
+      type: typeof FETCH_PENDING_PERMISSION_REQUEST_FAILURE;
+      error: string;
+    }
+  | {
+      type: typeof REQUEST_PARENTAL_PERMISSION_PERFORM;
+    }
+  | {
+      type: typeof REQUEST_PARENTAL_PERMISSION_SUCCESS;
+      parentalPermissionRequest: ParentalPermissionRequest;
+    }
+  | {
+      type: typeof REQUEST_PARENTAL_PERMISSION_FAILURE;
+      error: string;
+    };
+
+export interface ParentalPermissionRequestState {
+  action?: string | null;
+  isLoading: boolean;
+  parentalPermissionRequest?: ParentalPermissionRequest | null;
+  error?: string | null;
+}
+
+export default function parentalPermissionRequestReducer(
+  state: ParentalPermissionRequestState,
+  action: ParentalPermissionRequestAction
+): ParentalPermissionRequestState {
+  switch (action.type) {
+    case FETCH_PENDING_PERMISSION_REQUEST_PERFORM:
+      return {
+        ...state,
+        action: action.type,
+        isLoading: true,
+        error: null,
+      };
+    case FETCH_PENDING_PERMISSION_REQUEST_SUCCESS:
+      return {
+        ...state,
+        action: action.type,
+        isLoading: false,
+        parentalPermissionRequest: action.parentalPermissionRequest,
+      };
+    case FETCH_PENDING_PERMISSION_REQUEST_FAILURE:
+      return {
+        ...state,
+        action: action.type,
+        isLoading: false,
+        error: action.error,
+      };
+    case REQUEST_PARENTAL_PERMISSION_PERFORM:
+      return {
+        ...state,
+        action: action.type,
+        isLoading: true,
+        error: null,
+      };
+    case REQUEST_PARENTAL_PERMISSION_SUCCESS:
+      return {
+        ...state,
+        action: action.type,
+        isLoading: false,
+        parentalPermissionRequest: action.parentalPermissionRequest,
+      };
+    case REQUEST_PARENTAL_PERMISSION_FAILURE:
+      return {
+        ...state,
+        action: action.type,
+        isLoading: false,
+        error: action.error,
+      };
+    default:
+      return state;
+  }
+}
+
+export const fetchPendingPermissionRequest = (
+  dispatch: Dispatch<ParentalPermissionRequestAction>
+) => {
+  dispatch({type: FETCH_PENDING_PERMISSION_REQUEST_PERFORM});
+
+  fetch(FETCH_PENDING_PERMISSION_REQUEST_URL, {
```

Comment:
nit: could probably clean up this function and the one below it by using async/await instead of promise chaining.

Reply:
Are you suggesting we use async/await functions instead of more common for React the Redux/Reducer approach?

Reply:
Is the Redux/Reducer approach incompatible with async functions?

Reply:
My comment was just regarding convention - when possible, favor async/await over promise chaining or old school callbacks. If it isn't possible because of some nuance of Redux, feel free to ignore

Reply:
Oh, I see... https://github.com/code-dot-org/code-dot-org/pull/58337/commits/8451021fb833b9732259a9e80c89ac1b20f2e721